src/tequila/grouping/binary_rep.py

```diff
@@ -230,23 +232,54 @@ def anti_commutativity_matrix(self):
         gram = np.block([[np.zeros((n,n)), np.eye(n)], [np.eye(n), np.zeros((n,n))]])
         return matrix @ gram @ matrix.T % 2
 
-    def commuting_groups(self, method='rlf'):
+    def commuting_groups(self, method='rlf', condition="fc", overlap_aux=None):
         """
+        Notes
+        ----------
         Return the partitioning of the hamiltonian into commuting groups.
+
+        Parameters
+        ----------
+        method: str: The method used to group the commutative terms together. In the case of overlapping methods,
+        the coefficient for a given Pauli product term is split into several fragments optimally.
+        condition: str: Commutativitiy condition. Either fully commuting (fc) or qubit-wise commuting (qwc).
+        overlap_aux: class OverlappingAuxiliary: class contains the dictionary of covariances and the number of 
+        iterations in the overlapping method.
+
+        Returns
+        ----------
         List of BinaryHamiltonian's
         """
+        def method_class(method, condition):
+            """
+            Return the class that the method belongs to: One from Minimum clique cover (mcc)
+            and Greedy grouping algorithms. 
+            """
+            if (method == 'lf' or method == 'rlf'): 
+                mc = 'mcc' 
+                if condition != "fc": raise TequilaException(f"lf and rlf can only return fully commuting fragments.")
```

Comment:
Good for the user is to explain a bit what went wrong:
e.g.add something in the exception message like:
"combination of method={} and condition={} is not valid. E.g. lf and rlf can only return fully commuting fragments, so condotion needs to be condition=fc".format(method,condition)

Same for the other exceptions